### ISSUE TITLE
IAM Policy: Additional Permission required

### ISSUE THREAD

**Issue by A-Serr** [yxliaoyx/AWS-OnDemand-Server-Discord-Interaction#4]
Error: 
` "User: arn:aws:iam::XXXXXXXXXXX:user/discord_manager is not authorized to perform: ecr:PutLifecyclePolicy on resource: arn:aws:ecr:us-east-1:XXXXXXXXXXXX:reposit 
ory/cdk-hnb659fds-container-assets-XXXXXXXXXXXXXX-us-east-1 because no identity-based policy allows the ecr:PutLifecyclePolicy action (Service: Ecr, Status Code: 400, Request ID: 7795f459-ab81-4469-80df- 
73de18fb6dfc)" (RequestToken: ae2008b3-363e-d97c-debe-60536bdc416f, HandlerErrorCode: GeneralServiceException)
`

Resolved by adding `ecr:PutLifecyclePolicy` to IAM policy.

**Comment by 314pies** [comment]
Hm... AFAIK, this project doesn't use ECR?
Can you provide more details on what you were trying to setup?

**Comment by JamalMahroof** [comment]
Facing the same issue @314pies 

In the CF template on stack it is the [ContainerAssetsRepository] resource causing the issue and trying to use ECR. I believe something is dependent upon it, maybe a cdk thing?

docs/policies.json already contains the below, needs `ecr:PutLifecyclePolicy` adding to it. Lines 36-43

`"Effect": "Allow",
"Action": [
"ecr:SetRepositoryPolicy",
"ecr:GetLifecyclePolicy",
"ecr:PutImageScanningConfiguration",
"ecr:DescribeRepositories",
"ecr:CreateRepository",
"ecr:DeleteRepository"`

**Comment by yxliaoyx** [comment]
@JamalMahroof 

Where do you run the script? On EC2?

**Comment by JamalMahroof** [comment]
@yxliaoyx Yes on EC2

**Comment by 314pies** [comment]
@JamalMahroof 
Did this happen at the first time when you run the script? 

I'm suspecting it's some permission required for CDK bootstrapping but I haven't had a chance to verify it.

**Comment by JamalMahroof** [comment]
@314pies 

Yeah first time running the script. I believe cdk creates an ECR asset repo. Something I’ve seen in other projects recently. And needs the permission for putlifecyclepolicy on the repo it creates

Feel free to validate
